Update sidebar

duduBTW opened the pull request:
## A lot of UI updates.

- Update sidebar to match the new Figma design.
- Update song detail container design.
- Update filters design.
- Add support for multiple popovers opening one on top of another.
- Add new roving focus group function.
- Add new tabs component.
- Update song item context menu to use the `List` component.

https://github.com/user-attachments/assets/150c584b-184a-4265-afe2-bf293cac6d93

Tnixc commented:
<img width="82" alt="image" src="https://github.com/user-attachments/assets/d512ed0a-1fd3-4595-a4aa-cda9ad4c1970">


I don't think the border here looks good.

---

<img width="220" alt="image" src="https://github.com/user-attachments/assets/fa5929d3-f6b7-4182-b957-b50b54e42587"><img width="220" src="https://github.com/user-attachments/assets/46ccb9c1-3c18-4d0d-becd-53bb3d6cf7e6">


Context menu is not consistent with master.

<img width="201" alt="image" src="https://github.com/user-attachments/assets/81087d03-e3ad-4d48-a6d8-1ae3024bb9af">

---

![image](https://github.com/user-attachments/assets/d1da459e-087b-4fbe-af7b-1693f64181a4)


We'd want to align the buttons to the top on mac. I also think the sidebar button should be scaled down. (20px or 16px, same as settings button icon)

brw commented:
Very nitpicky but I think the close button on hover for Windows/Linux should be a much darker shade of red compared to what it is here.

Also preferably we'd have a different icon for un-maximizing (the two overlaid squares) like I mentioned before but since Lucide doesn't have an icon for it it's whatever I guess. Maybe we'll create our own at some point.

duduBTW commented:
> We'd want to align the buttons to the top on mac

We can't do that because we need a gap on the top for when the sidebar is collapsed, also to align with the queue button on the right.

<img width="1423" alt="image" src="https://github.com/user-attachments/assets/34cceb33-5e8c-4164-936b-dd1386e95b20">

duduBTW commented:
> Very nitpicky but I think the close button in the window controls for Windows/Linux should be a much darker shade of red on hover compared to the regular bg-red we have.

Update the hover color to be darker (looks like the default windows red now).

> Also preferably we'd have a different icon for un-maximizing (the two overlaid squares) like I mentioned before but since Lucide doesn't have an icon for it it's whatever I guess. Maybe we'll create our own at some point.

I think keeping it as a square it's fine for now, we can update it if we get a good "un-maximizing" icon.

duduBTW commented:
> <img alt="image" width="82" src="https://private-user-images.githubusercontent.com/85466117/382425276-d512ed0a-1fd3-4595-a4aa-cda9ad4c1970.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2NjQyNzIsIm5iZiI6MTczMDY2Mzk3MiwicGF0aCI6Ii84NTQ2NjExNy8zODI0MjUyNzYtZDUxMmVkMGEtMWZkMy00NTk1LWE0YWEtY2RhOWFkNGMxOTcwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAzVDE5NTkzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVlYWI4NTA1NWE5YTBiMjk5YjVkZWI5MTE0MDk3YzQzOGI1ZjhmODNiODk4MDcyYTUyNjhlYzg0NDM0NTY5NDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zzsot3GpxFTLAb1nyT2Rjb2ZRSvYKiOmjsPjMnjpYoY">

> I don't think the border here looks good.

Removed the border

---

> <img alt="image" width="220" src="https://private-user-images.githubusercontent.com/85466117/382425483-fa5929d3-f6b7-4182-b957-b50b54e42587.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2NjQyNzIsIm5iZiI6MTczMDY2Mzk3MiwicGF0aCI6Ii84NTQ2NjExNy8zODI0MjU0ODMtZmE1OTI5ZDMtZjZiNy00MTgyLWI5NTctYjUwYjU0ZTQyNTg3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAzVDE5NTkzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUwZWNlOWY1MjVlOWVkYzAwZWMwODNkNDE2YmUyOWJiYzNjN2EwZTA5Y2VkYmM0YTc5YmQ5YjVjZDJjYTIzNmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xEEQDtRfJX8bJMN07tB8dX6FyLwOKN-y9uQw4Uozr3Y"><img alt="" width="220" src="https://private-user-images.githubusercontent.com/85466117/382425592-46ccb9c1-3c18-4d0d-becd-53bb3d6cf7e6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2NjQyNzIsIm5iZiI6MTczMDY2Mzk3MiwicGF0aCI6Ii84NTQ2NjExNy8zODI0MjU1OTItNDZjY2I5YzEtM2MxOC00ZDBkLWJlY2QtNTNiYjNkNmNmN2U2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAzVDE5NTkzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU5ODFjOWY3NzI3NGZmYjNjYzRjNjQ0YzRlYmFlZDgyN2NjYWVmM2FjNmEwYjBjMGEzNDk0ZTQxZTg0MjIyYzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.uod-LgKdhe_Vp4GdS15-mjGzkWSt9MtyRjJtfZjk774">

> Context menu is not consistent with master.

Updated the context menu so it has all the options and it is not hovered by default

---
 
> <img alt="image" width="201" src="https://private-user-images.githubusercontent.com/85466117/382425766-81087d03-e3ad-4d48-a6d8-1ae3024bb9af.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2NjQyNzIsIm5iZiI6MTczMDY2Mzk3MiwicGF0aCI6Ii84NTQ2NjExNy8zODI0MjU3NjYtODEwODdkMDMtZTNhZC00ZDQ4LWE2ZDgtMWFlMzAyNGJiOWFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAzVDE5NTkzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkNzliMDEzYjkxNjc0YzhiMWU4ZTBkNzIzOWNiNGZkYmUzYTZjNmZiNWZkOTgxODUwMTRlNDhjYTIyMDJjYWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vzvqO3VhxLpSgkhKHT0aishX3FO9V5WXz32PSN20YOc">

> We'd want to align the buttons to the top on mac. I also think the sidebar button should be scaled down. (20px or 16px, same as settings button icon)

I don't think we should decrease the spacing, it keeps it more consistent with the queue button and between windows/mac.
It also gives it more space to breathe, you can notice it is still way too close to the top when you hover the button

---

> <img alt="image" width="413" src="https://private-user-images.githubusercontent.com/85466117/382426084-dc7a6baa-465c-4993-b7f3-2641d802c917.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2NjQyNzIsIm5iZiI6MTczMDY2Mzk3MiwicGF0aCI6Ii84NTQ2NjExNy8zODI0MjYwODQtZGM3YTZiYWEtNDY1Yy00OTkzLWI3ZjMtMjY0MWQ4MDJjOTE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAzVDE5NTkzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM4ODcxOTRhZDg4NmY0YjBmNDM1YjZkMGQ5ODA0MGQ4Mzg0MjU4ZGE0ZDcyNTkxNjYyNDc5YWRlYmRmODgyYTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.96VnpiBESLcqutpMuOpulITLPH1shk-5XDEeoq9kxgE">

> Double outline on focus.

Fixed that.

@Tnixc

Tnixc commented:
Thanks for fixing all that!

hrfarmer reviewed (changes requested):
code and design of this looks good to me, however i strongly feel that the scale of the ui across the board needs to be scaled down, it looks like it's scaled 125% compared to nearly any other desktop app

i think this is a decent example, if you compare the scale of osu!radio at default zoom to zen
<img width="1156" alt="image" src="https://github.com/user-attachments/assets/4db9a698-1950-41ab-ad3c-7e84a119b416">

if i `CMD + -` osu!radio twice, i think it looks more reasonable
<img width="1157" alt="image" src="https://github.com/user-attachments/assets/dda9e9db-d0e8-4f48-8f6a-753d10f9b618">

if people disagree then i'll let it go, but i really think it should be considered

brw commented:
I'm not opposed to scaling some stuff down but probably better to do that in a different PR

brw commented on src/renderer/src/scenes/main-scene/Sidebar.tsx:
```diff
@@ -0,0 +1,124 @@
+      </div>
+
+      <Show when={sidebarExpanded()}>
+        <ResizablePanel.DragHandler class="absolute -right-4 lg:-right-6 top-1 lg:pt-2 lg:pb-4" />
```
`top-1` here causes a scrollbar to be shown on the entire app window on Linux. It also doesn't seem necessary to offset it as without it's already perfectly centered for me. Different on macOS maybe?

hrfarmer replied:
yeah didn't see any scrollbar on mac

brw replied:
I meant more like is the `top-1` specifically to make it look good on macOS because of the different title bar

hrfarmer replied:
no i just realized i missed that commit and didn't pull it 😅

brw replied:
oh lol gotcha

hrfarmer commented:
> I'm not opposed to scaling some stuff down but probably better to do that in a different PR afterwards

well i brought it up here since this pr is already touching like all of the ui code, but if you think it should be addressed in a different pr then that's fair

brw commented:
It's about time this gets merged and scaling stuff down is a pretty significant change to the overall design so that's better suited as a separate conversation I'd say
